xamarin/Xamarin.Forms
# [Visual] Visual xaml and Visual registrations

### Description of Change ###

- Add a type converter into core.design that's able to display available visuals
- Add a public method on Visual Type Converter in case people need to register there own names to an IVisual. This should help if there are visual collisions or they just want to call it something else
- Added registration of Visual types to converter during assembly scanning processes


![image](https://user-images.githubusercontent.com/5375137/52991964-19296900-33cc-11e9-9707-e4521347ded3.png)


### API Changes ###

Added:
 - VisualTypeConverter.Register(string visualKey, IVisual visual)
 

### Platforms Affected ### 


- Core/XAML (all platforms)

### Testing Procedure ###
- to test xaml changes you have to take the nuget and install it into a project then make sure you get Visual options when adding Visual to elements

### PR Checklist ###
